### PR TITLE
added: allow vfs-addons to populate network setup dialog

### PR DESCRIPTION
**Pull request by notspiff**
see title.

### CODE REVIEW

File: xbmc/addons/VFSEntry.cpp [hunk 1]
```diff
@@ -134,15 +134,30 @@ class CVFSURLWrapper
     std::vector<std::string> m_strings;
 };
 
+CVFSEntry::ProtocolInfo::ProtocolInfo(BinaryAddonBasePtr addonInfo)
+  : supportPath(addonInfo->Type(ADDON_VFS)->GetValue("@pathSupport").asBoolean()),
+    supportUsername(addonInfo->Type(ADDON_VFS)->GetValue("@usernameSupport").asBoolean()),
+    supportPassword(addonInfo->Type(ADDON_VFS)->GetValue("@passwordSupport").asBoolean()),
+    supportPort(addonInfo->Type(ADDON_VFS)->GetValue("@portSupport").asBoolean()),
+    supportBrowsing(addonInfo->Type(ADDON_VFS)->GetValue("@browsingSupport").asBoolean()),
```
Comment (xhaggi) [inline xbmc/addons/VFSEntry.cpp]: Should we use the same wording like in the struct? `supportPath` etc

File: xbmc/addons/VFSEntry.cpp [hunk 2]
```diff
@@ -134,15 +134,30 @@ class CVFSURLWrapper
...
+    defaultPort(addonInfo->Type(ADDON_VFS)->GetValue("@defaultPort").asInteger()),
+    type(addonInfo->Type(ADDON_VFS)->GetValue("@protocols").asString()),
+    label(addonInfo->Type(ADDON_VFS)->GetValue("@labelID").asInteger())
```
Comment (xhaggi) [inline xbmc/addons/VFSEntry.cpp]: Here too, `label` instead of `labelID`

File: xbmc/network/GUIDialogNetworkSetup.cpp [hunk 3]
```diff
@@ -226,6 +226,16 @@ void CGUIDialogNetworkSetup::InitializeSettings()
   m_protocols.emplace_back(Protocol{true, true, true, true, false, 22, "sftp", 20260});
 #endif
 
+  for (const auto& addon : CServiceBroker::GetVFSAddonCache().GetAddonInstances())
```
Comment (xhaggi) [inline xbmc/network/GUIDialogNetworkSetup.cpp]: should we check for `CServiceBroker::IsBinaryAddonCacheUp()` before access addon cache?

File: xbmc/addons/VFSEntry.cpp (hunk 1 above)
Comment (notspiff) [reply xbmc/addons/VFSEntry.cpp]: fine.

File: xbmc/network/GUIDialogNetworkSetup.cpp (hunk 3 above)
Comment (notspiff) [reply xbmc/network/GUIDialogNetworkSetup.cpp]: done.

File: xbmc/addons/VFSEntry.cpp (hunk 2 above)
Comment (notspiff) [reply xbmc/addons/VFSEntry.cpp]: done.

**Review by xhaggi — APPROVED** [review]
Thx

**Comment by notspiff** [comment]
jenkins build this please

**Pull request merged**